src/main/kotlin/org/stellar/walletsdk/util/Sep24InteractiveFlow.kt

```diff
@@ -0,0 +1,139 @@
+package org.stellar.walletsdk.util
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import okhttp3.OkHttpClient
+import org.stellar.sdk.Network
+import org.stellar.sdk.Server
+import org.stellar.walletsdk.*
+
+/**
+ * Interactive flow for deposit and withdrawal using SEP-24.
+ *
+ * @param type chosen flow (deposit or withdrawal)
+ * @param accountAddress Stellar address of the account, used for authentication and by default for
+ * depositing or withdrawing funds
+ * @param fundsAccountAddress optional Stellar address of the account for depositing or withdrawing
+ * funds, if different from the account address
+ * @param homeDomain home domain of the anchor
+ * @param assetCode Asset code to deposit or withdraw
+ * @param memoId optional memo ID to distinguish the account
+ * @param clientDomain optional domain hosting stellar.toml file containing `SIGNING_KEY`
+ * @param extraFields Additional information to pass to the anchor
+ * @param walletSigner interface to define wallet client and domain (if using `clientDomain`)
+ * signing methods
+ * @param anchor instance of the [Anchor]
+ * @param server Horizon [Server] instance
+ * @param network Stellar [Network] instance
+ * @param httpClient HTTP client
+ *
+ * @return response object from the anchor
+ *
+ * @throws [AssetNotAcceptedForDepositException] if asset is not accepted for deposits
+ * @throws [AssetNotEnabledForDepositException] if asset is not enabled for deposits by the anchor
+ * @throws [AssetNotAcceptedForWithdrawalException] if asset is not accepted for withdrawals
+ * @throws [AssetNotEnabledForWithdrawalException] if asset is not enabled for withdrawals by the
+ * anchor
+ * @throws [NetworkRequestFailedException] if network request fails
+ */
+suspend fun interactiveFlow(
+  type: InteractiveFlowType,
+  accountAddress: String,
+  fundsAccountAddress: String? = null,
+  homeDomain: String,
+  assetCode: String,
+  memoId: String? = null,
+  clientDomain: String? = null,
+  extraFields: Map<String, Any>? = null,
+  walletSigner: WalletSigner,
+  anchor: Anchor,
+  server: Server,
+  network: Network,
+  httpClient: OkHttpClient,
+): InteractiveFlowResponse {
+  return CoroutineScope(Dispatchers.IO)
+    .async {
+      val sep24RequiredFields =
+        listOf(
+          StellarTomlFields.SIGNING_KEY.text,
+          StellarTomlFields.TRANSFER_SERVER_SEP0024.text,
+          StellarTomlFields.WEB_AUTH_ENDPOINT.text
+        )
+      val toml = StellarToml(homeDomain, server, httpClient)
+      val tomlContent = toml.getToml()
```

Comment:
We should cache toml contents somehow, because it doesn't make sense to fetch the toml file every time we want to make a deposit request. Ideally we cache it on the `Anchor` class.

Reply:
:100: I thought the same once I started working on the status update polling. I'm not yet sure how to best handle that in Kotlin, but we will figure it out in the status update PR.

Reply:
How does the transaction status polling logic relate to caching the toml or auth token?